### ISSUE TITLE
[Feature]: `uv`-managed environments should distinguish between required and optional dependencies

### ISSUE THREAD

**Issue by etienneschalk** [etienneschalk/scientific-data-viewer#89]
### Feature Category

Accessibility

### Priority

Low - Nice to have

### Feature Description

Add more granularity for dependency management with uv-management venvs

### Problem Statement

Currently when creating the extension's own environment via uv, everything is installed. It results in a large venv (order of magnitude 1GB)

### Proposed Solution

Let the user choose the dependencies to install when creating an environment, eg via a Quick Pick with checkboxes

The "click on configuration" should either trigger a quick pick (might not be wanted, bothering a user changing settings with a dialog is probably not the best), or just keep the current behaviour of installing evertyhing by default. 

Maybe we can consider a user checking the checkbox for the first time just wants it to work, and a user manually creating an env might want more granularity. In both cases the Getting Started guide should be updated to explain how to selectly install dependencies

An already created uv-managed env should also be updatable with new dependency, eg via "Add or Remove dependencies" quick pick option

### Use Cases

Reducing environment sizes, not everyone is interested in all formats

### Code of Conduct

- [x] I have searched existing issues and this is not a duplicate
- [x] I have read and agree to follow the [Contributing Guidelines](CONTRIBUTING.md)
